### PR TITLE
Support custom rules for read-only mode

### PR DESCRIPTION
**Pull request by korjek**
When dashboard is started with read-only role, CRDs are not shown in UI due to permissions issue.
This PR adds ability to provide custom rules for read-only role

### CODE REVIEW

**Comment by korjek** [comment]
@desaintmartin sorry for the late reply, it fell out of my radar.

>Wouldn't it be better to add the ability in this case to add in the clusterrole itself the missing permission?
Not sure that I got the idea. Could you please explain a bit more?

>in the case where someone wants to have its own customized rbacs, it's probably better to do it outside of the chart
If someone uses helm to deploy everything, that would mean (if I got your idea correct) creating a custom in-house chart and using it to create role with custom permissions for k8s dashboard. I don't think this is a good way to go.

**Comment by korjek** [comment]
> Instead of setting a new option and telling users to add CRD permissions in extra RBAC to have a working dashboard, shouldn't it be done by default?

I don't think it's possible, as every CRD has it's own apiGroup.
So we with the PR I propose we have the next lines in values.yaml to be able to see `sparkapplications` in k8s dashboard:
```
    clusterReadOnlyRoleAdditionalRules:
      - apiGroups:
        - apiextensions.k8s.io
        resources:
        - customresourcedefinitions
        verbs:
        - get
        - list
        - watch
      - apiGroups:
        - sparkoperator.k8s.io
        resources:
        - sparkapplications
        - sparkapplications/status
        - scheduledsparkapplication
        - scheduledsparkapplication/status
        verbs:
        - get
        - list
        - watch
```

**Comment by korjek** [comment]
@desaintmartin thoughts?

**Comment by korjek** [comment]
@desaintmartin could we proceed this way? or what other options do we have instead of forking?

**Comment by korjek** [comment]
@shu-mutou maybe you can help here?

**Comment by desaintmartin** [comment]
Sorry for the huge delay. Indeed this makes sense for CRDs.

File: aio/deploy/helm-chart/kubernetes-dashboard/Chart.yaml [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 apiVersion: v2
 name: kubernetes-dashboard
-version: 5.5.2
+version: 5.5.3
```
Comment (desaintmartin) [inline aio/deploy/helm-chart/kubernetes-dashboard/Chart.yaml:17]:
```suggestion
version: 5.6.0
```
(after a rebase)

**Comment by korjek** [comment]
@desaintmartin thank you!
Resolved conflicts

**Pull request merged**